### ISSUE TITLE
arch specific version still errors

### ISSUE THREAD

**Issue by boomam** [themattman/mongodb-raspberrypi-docker#1]
...With it being unofficial, I dont expect that this would be looked into with any priority, but to flag this up - 
Build successfully runs, container successfully pulls, deployment however fails with the same issues as normal mongo in its log - 

```
WARNING: Using unofficial community build for Raspberry Pi which was purpose-built to overcome the micro-architecture limitations.
  see also https://github.com/docker-library/mongo/issues/485#issuecomment-970864306
  see also https://en.wikichip.org/wiki/arm/armv8#ARMv8_Extensions_and_Processor_Features
  see https://jira.mongodb.org/browse/SERVER-55178
WARNING: MongoDB 5.0+ requires ARMv8.2-A or higher, and your current system does not appear to implement any of the common features for that!
  see https://github.com/themattman/mongodb-raspberrypi-binaries
```
Specifically, using v6.1.0-rc4.

**Comment by themattman** [comment]
> deployment however fails with the same issues as normal mongo in its log

What is the return code of `mongod`? What do the `mongod` logs say?

The log line you quoted from Docker is expected on successful execution. It is a warning, part of which I authored noting that these are unofficial and for the Pi. For the next release in this repo I have removed the second warning.

**Comment by boomam** [comment]
Hi,
I'll keep an eye out for the second release and re-test then - IIRC (it was a while ago i was testing this), the container was crashing after the above error regardless, netting the same end result as the 'official' container after it fails.  
Thanks!

**Comment by themattman** [comment]
@boomam give one of the new releases a spin.

**Comment by Antares84** [comment]
Attempting to run r7.0.1, docker import, Cannot create container for service mongo: No command specified

**Comment by themattman** [comment]
I am not very familiar with Docker, but could you try `docker load` instead of `docker import`?

**Comment by boomam** [comment]
> @boomam give one of the new releases a spin.

@themattman 
Apologies for the delay.
Tested today, _appears_ to start. Need to do a little more testing.

Created a PR [here](https://github.com/themattman/mongodb-raspberrypi-docker/pull/3) to add some basic build steps, as i saw a question in the other issue.

**Comment by boomam** [comment]
Just tested with 6.0.11 (previous test was with 7.0.3), built and replaced the image/release on a Graylog stack.  

**Fundamentally, it looks like Mongo comes up**   

However, for those on old MongoDb versions (such as those using Graylog on a Raspberry Pi 4), i had to do a few upgrade/migration steps -  
&nbsp;  
The TL:DR is that, as-is, the container constantly restarts with an error similar to the below -  

```sh
ctx=initandlisten msg=Wrong mongod version attr={"error":"UPGRADE PROBLEM: Found an invalid featureCompatibilityVersion document (ERROR: Location4926900: Invalid featureCompatibilityVersion document in admin.system.version: { _id: \"featureCompatibilityVersion\", version: \"4.2\" }. 
See https://docs.mongodb.com/master/release-notes/5.0-compatibility/#feature-compatibility. :: 
caused by :: Invalid feature compatibility version value, expected '5.0' or '5.3' or '6.0. See https://docs.mongodb.com/master/release-notes/5.0-compatibility/#feature-compatibility.). 
If the current featureCompatibilityVersion is below 5.0, see the documentation on upgrading at https://docs.mongodb.com/master/release-notes/5.0/#upgrade-procedures."}
```

After a little trial and error, and verifying what upgrade steps were needed [here](https://www.mongodb.com/docs/upcoming/reference/command/setFeatureCompatibilityVersion/#default-values), i performed the following steps -  

1. I copied the mongo data to an x86 host, and deployed a [4.4.26-rc0-focal](https://hub.docker.com/layers/library/mongo/4.4.26-rc0-focal/images/sha256-944caba8eba9adc561474755eec88aae1557f92790985248e88e3b44cb398aa0?context=explore) image, pointing at said data.

2. I then i used `docker exec -it mongo /bin/sh` to get to its CLI, and ran the following commands -  
```sh
mongo
use admin
```
This logged me into the mongo CLI is admin.  

3. I checked the current featureCompatibiltyVersion level -  
`db.runCommand({getParameter: 1, featureCompatibilityVersion: 1})`  
Followed by updating it to v5.0 -  
`db.adminCommand({ setFeatureCompatibilityVersion: "4.4" })`  
Then verified it stuck with the check command again -  
`db.runCommand({getParameter: 1, featureCompatibilityVersion: 1})`  
&nbsp;  

4. Next, i updated the x86 container to a [5.0.23-rc0-focal](https://hub.docker.com/layers/library/mongo/5.0.23-rc0-focal/images/sha256-60eea50b7b1dfae9fa589a1f6394ee83ad080ab246f176a268da3e4dd23e8435?context=explore) image.  
&nbsp;  
Then performed the same steps again, this time updating the version to `5.0`.  
&nbsp;  

5. Following this, i copied the data back to the Raspberry Pi, corrected permissions, started the 6.0.11 image from here, and....it stays up!  

Then to make it easier for myself, in future, i did one last version update to v6.0, should i ever upgrade to v7.x+.  

&nbsp;  
In my case, starting Graylog v5 still was not successful, as it needed some [additional steps](https://go2docs.graylog.org/5-2/upgrading_graylog/upgrading_graylog_in_docker.htm).  
&nbsp;  
Hopefully this helps people upgrade!

**Comment by themattman** [comment]
Thank you for the detailed steps, @boomam! I have seen at least one other user who is using these images/binaries for Graylog. Adding a wiki page on the repo might be a good next step for common applications (e.g. Unifi). Yes, it looks like you ran into the [one-upgrade-at-a-time](https://www.mongodb.com/docs/v7.0/release-notes/7.0-upgrade-standalone/#upgrade-version-path) requirement from MongoDB. Glad you got it sorted to be able to jump from `4.4` up to `6.x`!

I'll close this issue for now as it appears that the docker images themselves are sound.
